test/utils/shallow.js

```diff
@@ -0,0 +1,18 @@
+import { ShallowWrapper } from 'enzyme'
+
+const meetsName = (wrapper, name) => name === wrapper.type().name
+
+const dive = wrapper => {
+  if (meetsName(wrapper, 'refHOC')) wrapper = wrapper.dive()
+  if (meetsName(wrapper, 'Ref')) wrapper = wrapper.dive()
+
+  return wrapper
+}
+
+class Wrapper extends ShallowWrapper {
+  shallow(options) {
+    return this.single('shallow', n => dive(new Wrapper(n, null, options)))
+  }
+}
+
+export default (...args) => dive(new Wrapper(...args))
```

Comment:
@levithomason I need you there. 

After I injected `withRef` HOC to `getElementType` I received many broken tests. Half of then fixed with the code above, its idea to `dive` automatically when `shallow` is called.

Another half shoud be fixed with overriding of the `shallow` method on `ShallowWrapper`, but the inheritance **doesn't** work. WTF? Can you explain what I'm doing wrong? I think that I'm missing something basic, but at now it destroys my mind. `phantomjs` doesn't support neither `Proxy`, nor `__noSuchMethod__` :cry: 

Example test:

```jsx
 shallow(<Confirm cancelButton='foo' />)
   .find('Button')
   .first()
    .shallow()
    .should.have.text('foo')
```

```js
expected <Button /> to have text 'foo', but it has '<refHOC />'
```

Reply:
```js
const shallowMethod = ShallowWrapper.prototype.shallow

ShallowWrapper.prototype.shallow = function (...args) {
  return dive(shallowMethod.apply(this, ...args))
}
```

I found way to solve this, looks ugly, but works. Open for better solution